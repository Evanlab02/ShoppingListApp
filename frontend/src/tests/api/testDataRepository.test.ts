import { beforeEach, expect, test } from 'vitest'
import { DASHBOARD_CURRENT_MOCK, DASHBOARD_HISTORY_MOCK, DASHBOARD_RECENT_ITEMS_MOCK } from '../mock/dataRepoMocks'
import DataRepo from '../../api/dataRepository';

beforeEach(() => {
    fetchMock.resetMocks();
})

test("Data repository returns current dashboard data.", async () => {
    fetchMock.mockResponseOnce(JSON.stringify(DASHBOARD_CURRENT_MOCK));

    const repo = new DataRepo();
    const data = await repo.getDashboardCurrent();

    expect(data).toStrictEqual(DASHBOARD_CURRENT_MOCK);
    expect(data).toMatchSnapshot(DASHBOARD_CURRENT_MOCK);
    expect(data.total).toEqual(10);
    expect(data.total_price).toEqual(300);
    expect(data.budget_remaining).toEqual(150);
    expect(data.average_item_price).toEqual(80);
});

test("Data repository returns dashboard history.", async() => {
    fetchMock.mockResponseOnce(JSON.stringify(DASHBOARD_HISTORY_MOCK));

    const repo = new DataRepo();
    const data = await repo.getDashboardHistory();

    expect(data).toStrictEqual(DASHBOARD_HISTORY_MOCK);
    expect(data).toMatchSnapshot(DASHBOARD_HISTORY_MOCK);
});

test("Data repository returns recent items.", async() => {
    fetchMock.mockResponseOnce(JSON.stringify(DASHBOARD_RECENT_ITEMS_MOCK));

    const repo = new DataRepo();
    const data = await repo.getRecentItems();

    expect(data).toStrictEqual(DASHBOARD_RECENT_ITEMS_MOCK);
    expect(data).toMatchSnapshot(DASHBOARD_RECENT_ITEMS_MOCK);
});
