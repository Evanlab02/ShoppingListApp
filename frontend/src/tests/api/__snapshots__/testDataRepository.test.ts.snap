// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Data repository returns current dashboard data. 1`] = `
{
  "average_item_price": 80,
  "budget_remaining": 150,
  "total": 10,
  "total_price": 300,
}
`;

exports[`Data repository returns dashboard history. 1`] = `
{
  "data": [
    {
      "data": [
        65,
        59,
        80,
        81,
        56,
        55,
      ],
      "label": "Price",
    },
    {
      "data": [
        28,
        48,
        40,
        19,
        86,
        27,
      ],
      "label": "Budget",
    },
  ],
  "labels": [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
  ],
}
`;

exports[`Data repository returns recent items. 1`] = `
{
  "items": [
    {
      "name": "1",
      "price": 100,
    },
    {
      "name": "2",
      "price": 200,
    },
    {
      "name": "3",
      "price": 300,
    },
    {
      "name": "4",
      "price": 400,
    },
    {
      "name": "5",
      "price": 500,
    },
  ],
}
`;
